include/swift/AST/DiagnosticsSIL.def

```diff
@@ -269,6 +269,8 @@ ERROR(tf_op_misuse, none,
       "%0", (StringRef))
 NOTE(tf_op_misuse_note, none,
      "%0", (StringRef))
+ERROR(tf_op_extract_address_only,none,
+      "cannot extract TensorFlow result into an address-only type", ())
```

Comment:
"Address-only" is a SIL term, but diagnostics are to be shown to users. What user code would cause this error to occur? If it's not a user error, an assertion would be more appropriate. Otherwise, this shouldn't include any compiler terminology.

Reply:
Users can create code where the type is still address-only in the deabstraction scope. e.g. if they try to instantiate a dataset with a type that is still generic, or if they try to instantiate a dataset with a non-fixed-layout struct defined in a different module.

I could write a few specific errors for common cases ("cannot extract TensorFlow result into generic parameter", "cannot extract TensorFlow result into struct defined in other module"), and a more generic catch-all in case I missed anything ("cannot extract TensorFlow value into type %0"). WDYT?

Reply:
We chatted in person. Emitting a diagnostic without compiler terminology makes sense.

When the struct is not fixed-layout, we should reject packing. Could you add a diagnostic there as well?

Reply:
Done!

I think that the existing diagnostics catch non-fixed-layout when appropriate already (when the non-fixed-layout struct is defined in an external module, and resilience is enabled). I added a test case `packResultsToAggregate_toExternalStructNotFixedLayout` to make sure.